### PR TITLE
Remove URL fragments

### LINKED ISSUE #1090: Discard URL fragments when requesting annotations for a page
We don't treat this:
http://lappland.io/blog/2014/03/open-peer-review-as-annotations/

The same as this:
[http://lappland.io/blog/2014/03/open-peer-review-as-annotations/#fn:Diigo link feature](http://lappland.io/blog/2014/03/open-peer-review-as-annotations/#fn:Diigo link feature)

So, different annotations are showing at each.

### PR DESCRIPTION
**Pull request by csillag**
From now on, we don't store the URL fragment (#whatever) in the URL that we put into the annotation.
We also don't include it in the query URLs, when we load annotations.

Closes #1090.

We should still do #1537, though, to solve the problem of existing annotations.

### CODE REVIEW

File: h/static/scripts/annotator/plugin/fragmentselector.coffee [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Annotator plugin for creating the Fragment Selector
+class Annotator.Plugin.FragmentSelector extends Annotator.Plugin
+
+  pluginInit: ->
+
+    @Annotator = Annotator
```
Comment (tilgovi) [inline h/static/scripts/annotator/plugin/fragmentselector.coffee]: Not used.

File: h/static/scripts/annotator/plugin/fragmentselector.coffee [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    @anchoring = @annotator.anchoring
```
Comment (tilgovi) [inline h/static/scripts/annotator/plugin/fragmentselector.coffee]: Not used outside this method.

File: h/static/scripts/guest.coffee [hunk 3]
```diff
@@ -119,9 +120,27 @@ class Annotator.Guest extends Annotator
       # Announce the new positions, so that the sidebar knows
       this.plugins.Bridge.sync([highlight.annotation])
 
+  # Utility function to remove the hash part from a URL
+  _removeHash: (url) ->
+    url = new URL url
```
Comment (tilgovi) [inline h/static/scripts/guest.coffee]: This API is not the most portable: https://developer.mozilla.org/en/docs/Web/API/URL#Browser_compatibility
Can you conceive of an alternative or include a polyfill?

File: h/static/scripts/annotator/plugin/fragmentselector.coffee (hunk 1 above)
Comment (csillag) [inline h/static/scripts/annotator/plugin/fragmentselector.coffee]: > Not used.

Right. Removed.

File: h/static/scripts/annotator/plugin/fragmentselector.coffee (hunk 2 above)
Comment (csillag) [inline h/static/scripts/annotator/plugin/fragmentselector.coffee]: > Not used outside this method.

Removed.

File: h/static/scripts/guest.coffee (hunk 3 above)
Comment (csillag) [inline h/static/scripts/guest.coffee]: > This API is not the most portable: https://developer.mozilla.org/en/docs/Web/API/URL#Browser_compatibility

True; but we are already using it - see [here](https://github.com/hypothesis/h/blob/a8811acdc63b014acc82d712d3dc83c4e8b9197b/h/static/scripts/annotator/plugin/pdf.coffee#L287)

> Can you conceive of an alternative or include a polyfill?

I'll look for a polyfill. (I would hate to use some partial workaround when there is a proper API for doing something.)

Comment (csillag) [inline h/static/scripts/guest.coffee]: > Can you [...] include a polyfill?

How about [this one](https://github.com/inexorabletash/polyfill/blob/master/url.js), by @inexorabletash ? (Listed [here](https://github.com/Modernizr/Modernizr/wiki/HTML5-Cross-Browser-Polyfills).)

Comment (csillag) [inline h/static/scripts/guest.coffee]: Added polyfill in https://github.com/hypothesis/h/commit/4f46e9ff88455b651e8e039cbaee6ba589633b79.

Comment (tilgovi) [inline h/static/scripts/guest.coffee]: That looks good, thanks.

**Comment by csillag** [comment]
Any further comments? Can we merge this?

**Pull request merged**